### PR TITLE
Added Capabilities to provision MongoDB

### PR DESCRIPTION
**Pull request by ajithkvp**
**Description:**
Added Capabilities to provision MongoDB.
Included the following changes, 
1. Extended the current capabilities to include payload to provision MongoDB in GenerateProvisioningRequest module
2. Implemented Factory Pattern for generating Action Arguments based on DB type
3. Unit test cases were written and tested for the newly added functionality

**Testing:**
Tested this capability by applying the manifest for MongoDB. Was able to generate the provisioning request successfully in the NDB Era test Drive. It was also tested and verified by Manav Rajvanshi ( manav.rajvanshi@nutanix.com )

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ The NDB operator brings automated and simplified database administration, provis
 ![Proudly written in Golang](https://img.shields.io/badge/written%20in-Golang-92d1e7.svg)
 
 ---
-
+## Demo video
+[link](https://www.youtube.com/watch?v=CqZ7U9pxZ60)
```
Comment (manavrajvanshi) [inline README.md:16]: This should be removed.

Comment (ajithkvp) [reply README.md]: Done. Thanks.

**Comment by manavrajvanshi** [comment]
LGTM, Please merge the latest changes from `main`. @ajithkvp @nandinimundra00

File: test/ndb_api_helpers_test.go [hunk 2]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
 		}
 	}
 }
+
+func TestGetActionArgumentsByDatabaseType(t *testing.T) {
+	// Test with MySQL database type
+	mysqlArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MYSQL)
+	if mysqlArgs != nil {
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:612]: Should it be ` mysqlArgs == nil`?

File: test/ndb_api_helpers_test.go [hunk 3]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
...
+		t.Error("Expected non-nil, but got nil")
+	}
+
+	// Test with Postgres database type
+	postgresArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_POSTGRES)
+	if err != nil {
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:618]: same here. The error message says "t.Error("Expected non-nil, but got nil")"

If you  "got nil", wouldn't it imply `err == nil` evaluated to True?

File: api/v1alpha1/ndb_api_helpers.go [hunk 4]
```diff
@@ -261,3 +251,77 @@ func GenerateDeprovisionDatabaseServerRequest() (req *DatabaseServerDeprovisionR
...
+}
+
+func (m *MysqlActionArgs) GetActionArguments(dbSpec DatabaseSpec) []ActionArgument {
+	return []ActionArgument{}
+}
+
+func (p *PostgresActionArgs) GetActionArguments(dbSpec DatabaseSpec) []ActionArgument {
+	return []ActionArgument{
+		{
+			Name:  "proxy_read_port",
```
Comment (akshmish) [inline api/v1alpha1/ndb_api_helpers.go:278]: Consider defining these strings (eg listener_port) as constants in the constants file.

File: test/ndb_api_helpers_test.go [hunk 5]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
...
+		{Name: "enable_synchronous_mode", Value: "false"},
+		{Name: "auto_tune_staging_drive", Value: "true"},
+	}
+	if !reflect.DeepEqual(postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}), expectedPostgresArgs) {
+		t.Errorf("Expected %v, but got %v", expectedPostgresArgs, postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}))
+	}
+
+	// Test with MongoDB database type
+	mongodbArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MONGODB)
+	if err != nil {
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:634]: Same here, should this be "err == nil"?

File: test/ndb_api_helpers_test.go [hunk 6]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
...
+		{Name: "working_dir", Value: "/tmp"},
+		{Name: "db_user", Value: "test"},
+	}
+	if !reflect.DeepEqual(mongodbArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}), expectedMongodbArgs) {
+		t.Errorf("Expected %v, but got %v", expectedMongodbArgs, mongodbArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}))
+	}
+
+	// Test with unknown database type
+	unknownArgs, err := v1alpha1.GetActionArgumentsByDatabaseType("unknown")
+	if err == nil {
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:651]: err != nil?

File: test/ndb_api_helpers_test.go [hunk 7]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
 		}
 	}
 }
+
+func TestGetActionArgumentsByDatabaseType(t *testing.T) {
+	// Test with MySQL database type
+	mysqlArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MYSQL)
+	if mysqlArgs != nil {
+		t.Error("Expected non-nil, but got nil")
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:613]: The error message and the if-condition look off. Can you double check? I believe the same is true in couple of more places in the file.

File: test/ndb_api_helpers_test.go [hunk 8]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
...
+	}
+
+	// Test with Postgres database type
+	postgresArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_POSTGRES)
+	if err != nil {
+		t.Error("Expected non-nil, but got nil")
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:619]: but got nil implies err == nil evaluated to True, but the condition is "err != nil"

File: test/ndb_api_helpers_test.go [hunk 9]
```diff
@@ -604,3 +605,50 @@ func TestGenerateProvisioningRequestReturnsErrorIfSSHKeyIsEmpty(t *testing.T) {
...
+		{Name: "auto_tune_staging_drive", Value: "true"},
+	}
+	if !reflect.DeepEqual(postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}), expectedPostgresArgs) {
+		t.Errorf("Expected %v, but got %v", expectedPostgresArgs, postgresArgs.GetActionArguments(v1alpha1.DatabaseSpec{Instance: v1alpha1.Instance{DatabaseInstanceName: "test"}}))
+	}
+
+	// Test with MongoDB database type
+	mongodbArgs, err := v1alpha1.GetActionArgumentsByDatabaseType(v1alpha1.DATABASE_TYPE_MONGODB)
+	if err != nil {
+		t.Error("Expected non-nil, but got nil")
```
Comment (akshmish) [inline test/ndb_api_helpers_test.go:635]: "Expected non-nil, but got nil" implies "err == nil" evaluated as "True", but the condition is "err != nil"

File: test/ndb_api_helpers_test.go (hunk 6 above)
Comment (akshmish) [inline test/ndb_api_helpers_test.go:651]: Same here

File: test/ndb_api_helpers_test.go (hunk 2 above)
Comment (shenoypritika) [inline test/ndb_api_helpers_test.go:612]: Could you add the check for expected MYSQL arguments like you have done for postgres arguments below...

**Comment by manavrajvanshi** [comment]
The container build failed, Looks like a test has failed. Please fix the test before merging.

**Pull request closed without merging**